### ISSUE TITLE
Full Path Disclosure  

### ISSUE THREAD

**Issue by Winthan** [interledgerjs/ilp-kit#136]
`https://zar-ilp.hopebailie.com/ledger/accounts/"`

Requesting the account with an invalid parameter will cause an error page to be generated containing the path of the Ledger. 

Information obtained by exploiting this issue may aid an attacker in launching further attacks against a target server.

**Comment by sharafian** [comment]
Could you give an example URL that produces that output? By path of the ledger, do you mean on the filesystem?

**Comment by Winthan** [comment]
If you visited the pages, you will see 3 type of respond.

```
 https://zar-ilp.hopebailie.com/ledger/accounts/ (protected)
 https://zar-ilp.hopebailie.com/ledger/accounts/random ( this error is okay) 
 https://zar-ilp.hopebailie.com/ledger/accounts/admin'admin  (error with full path) 
```

The error message contains the full path of the Ledger. it might be a case of debug option. But for production,  it should not show actually. 

`{"id":"InvalidUriParameterError","message":"name is not a valid Identifier","validationErrors":[{"message":"String does not match pattern: ^([a-zA-Z0-9][a-zA-Z0-9-]{0,18}[a-zA-Z0-9]|([a-zA-Z0-9][a-zA-Z0-9-]{0,18}[a-zA-Z0-9]@)?((?!-)[a-zA-Z0-9-]{1,63}\\.)+(?!-)[a-zA-Z0-9-]{1,63})(?:\\:[0-9]{0,5})?$","params":{"pattern":"^([a-zA-Z0-9][a-zA-Z0-9-]{0,18}[a-zA-Z0-9]|([a-zA-Z0-9][a-zA-Z0-9-]{0,18}[a-zA-Z0-9]@)?((?!-)[a-zA-Z0-9-]{1,63}\\.)+(?!-)[a-zA-Z0-9-]{1,63})(?:\\:[0-9]{0,5})?$"},"code":202,"dataPath":"","schemaPath":"/pattern","subErrors":null,"stack":"Error\n    at new ValidationError (/home/adrian/ilp-kit/node_modules/tv4/tv4.js:1461:12)\n    at ValidatorContext.createError (/home/adrian/ilp-kit/node_modules/tv4/tv4.js:359:14)\n    at ValidatorContext.validateStringPattern (/home/adrian/ilp-kit/node_modules/tv4/tv4.js:869:15)\n    at ValidatorContext.validateString (/home/adrian/ilp-kit/node_modules/tv4/tv4.js:825:11)\n    at ValidatorContext.validateAll (/home/adrian/ilp-kit/node_modules/tv4/tv4.js:601:11)\n    at Object.validateMultiple (/home/adrian/ilp-kit/node_modules/tv4/tv4.js:1597:12)\n    at validate (/home/adrian/ilp-kit/node_modules/five-bells-ledger/node_modules/five-bells-shared/services/validate.js:29:20)\n    at Object.exports.validateUriParameter (/home/adrian/ilp-kit/node_modules/five-bells-ledger/node_modules/five-bells-shared/utils/request.js:21:26)\n    at Object.getResource (/home/adrian/ilp-kit/node_modules/five-bells-ledger/src/controllers/accounts.js:58:11)\n    at next (native)\n    at onFulfilled (/home/adrian/ilp-kit/node_modules/co/index.js:65:19)\n    at /home/adrian/ilp-kit/node_modules/co/index.js:54:5\n    at Object.co (/home/adrian/ilp-kit/node_modules/co/index.js:50:10)\n    at Object.toPromise (/home/adrian/ilp-kit/node_modules/co/index.js:118:63)\n    at next (/home/adrian/ilp-kit/node_modules/co/index.js:99:29)\n    at onFulfilled (/home/adrian/ilp-kit/node_modules/co/index.js:69:7)"}]}`

**Comment by sentientwaffle** [comment]
This should be fixed by `five-bells-ledger@19.4.1`
